src/hotspot/share/memory/arena.hpp

```diff
@@ -185,7 +185,9 @@ class Arena : public CHeapObj<mtNone> {
   // Fast delete in area.  Common case is: NOP (except for storage reclaimed)
   bool Afree(void *ptr, size_t size) {
 #ifdef ASSERT
-    if (ZapResourceArea) memset(ptr, badResourceValue, size); // zap freed memory
+    if (ZapResourceArea && ptr != NULL) {
```

Comment:
The pointer arithmetic on line 193 is UB if ptr is nullptr (though it's unlikely anyone checks), so either Afree has `ptr != nullptr` as an (unstated) precondition or there are potentially more problems than the memset.